main/s390-tools/bash.patch

```diff
@@ -0,0 +1,73 @@
+diff --git a/scripts/dbginfo.sh b/scripts/dbginfo.sh
+index bac62f0..3bed72a 100755
+--- a/scripts/dbginfo.sh
++++ b/scripts/dbginfo.sh
+@@ -1125,7 +1125,7 @@ exec 8>&1 9>&2 >"${LOGFILE}" 2>&1
+ trap emergency_exit SIGHUP SIGINT SIGTERM
+ 
+ pr_log_stdout ""
+-pr_log_stdout "Hardware platform     = $(uname -i)"
++pr_log_stdout "Hardware platform     = $(uname -m)"
+ pr_log_stdout "Kernel version        = ${KERNEL_VERSION}.${KERNEL_MAJOR_REVISION}.${KERNEL_MINOR_REVISION} ($(uname -r 2>/dev/null))"
+ pr_log_stdout "Runtime environment   = ${RUNTIME_ENVIRONMENT}"
+ pr_log_stdout ""
+diff --git a/zconf/chccwdev b/zconf/chccwdev
+index b1140c0..2c3ca6a 100755
+--- a/zconf/chccwdev
++++ b/zconf/chccwdev
+@@ -429,9 +429,9 @@ for BUSID in $BUSIDLIST; do
+ 				exit 1
+ 			fi
+ 			if [ "$FORCE" != "" ]; then
+-				echo $FORCE 2>&1 >$SYSPATH/$ONLINEATTR | PrintError
++				/bin/echo $FORCE 2>&1 >$SYSPATH/$ONLINEATTR | PrintError
+ 			else
+-				echo $ONLINE 2>&1 > $SYSPATH/$ONLINEATTR | PrintError
++				/bin/echo $ONLINE 2>&1 > $SYSPATH/$ONLINEATTR | PrintError
```

Comment:
Goodness. Those are some odd redirections!

Why is `/bin/echo` better here anyway?

Reply:
Not really odd. Redirecting stderr to stdout (your terminal) so any error from echoing to that /sys path would be caught by `PrintError` function. It could be rewritten as `echo $ONLINE > $SYSPATH/$ONLINEATTR |& PrintError`, if I understand the code correctly :D